### PR TITLE
feat: bentoml cli supoort import model from hf

### PR DESCRIPTION
**Pull request by xianml**
## What does this PR address?

bentoml-cli support import model from hf, like

`bentoml import https://huggingface.co/microsoft/phi-1_5`

test result
image

Fixes #(issue)

## Before submitting:

- [x] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [x] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [x] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/models.py [hunk 1]
```diff
@@ -115,6 +122,33 @@ def import_model(
...
+        diffuser_tags = [
+            "diffusers",
+            "stable-diffusion",
+            "diffusers:StableDiffusionPipeline",
+        ]
+        is_diffuser = False
+        for tag in diffuser_tags:
+            if tag in info.tags:
+                is_diffuser = True
+                break
```
Comment (frostming) [inline src/bentoml/models.py:145]:
```suggestion
        is_diffuser = any(tag in info.tags for tag in diffuser_tags)
```

File: src/bentoml/_internal/frameworks/transformers.py [hunk 2]
```diff
@@ -689,7 +690,11 @@ def save_model(
             DeprecationWarning,
         )
         pretrained_or_pipeline = pipeline
-
+    # construct pipeline from task_name and model-id
+    if isinstance(pretrained_or_pipeline, str):
+        pretrained_or_pipeline = transformers.pipeline(
+            task=task_name, model=pretrained_or_pipeline
+        )
```
Comment (aarnphm) [inline src/bentoml/_internal/frameworks/transformers.py:697]: I don't think we should load the pipeline in memory here. The whole purpose of importing API is that we don't load the model into memory

We should use `snapshot_download` instead.

File: src/bentoml/models.py [hunk 3]
```diff
@@ -115,6 +121,46 @@ def import_model(
     Returns:
         Model: the imported model
     """
+    parsedURL = urlparse(path)
+    if (
+        parsedURL.scheme == "http" or parsedURL.scheme == "https"
+    ) and parsedURL.netloc == "huggingface.co":
+        model_id = parsedURL.path.strip("/")
```
Comment (aarnphm) [inline src/bentoml/models.py:128]: I assume we are making a specific branch for huggingface here? Should we use the `bentoml.transformers.import_model` instead? 

Ofc once `bentoml.transformers.import_model` is ready.

**Comment by aarnphm** [comment]
Closed due to #4247

**Pull request closed without merging**